ionic_langchain/tool.py

```diff
@@ -41,21 +19,24 @@ def __init__(self, sdk: Optional[IonicSDK] = None):
 
     def query(
         self,
-        query_input: QueryInput,
+        query_input: str,
     ) -> Sequence[dict[str, Any]]:
-        if not query_input.query:
+        if not query_input:
             raise ValueError("query must not be empty")
         """
         :param query_input:  see QueryInput
         :return:
         """
-        query = query_input.query
+        split_query = query_input.split(",")
```

Comment:
Can you add a test for various valid and invalid permutations? This is pretty complex so want to make sure we're covered

Reply:
https://peps.python.org/pep-3132/ maybe valuable

Reply:
for testing, mock the request or better yet split this into a helper and just test that.